### PR TITLE
FLUID-5886: Fix for failure to deduplicate model listeners by namespace

### PR DESCRIPTION
**Pull request by amb26**
at a single applier. This rebalances the implementation of change events and standard events, with the former now implemented in terms of the latter. This introduces a new options-based API for event.addListener and an improved fluid.isPlainObject.

### CODE REVIEW

File: src/framework/core/js/Fluid.js [hunk 1]
```diff
@@ -1354,24 +1357,44 @@ var fluid = fluid || fluid_2_0_0;
...
                     return;
                 }
+                if (fluid.isPlainObject(listener, true)) {
+                    record = listener;
+                    listener = record.listener;
+                    namespace = record.namespace;
+                    priority = record.priority;
+                    listenerId = record.listenerId;
+                } else {
+                    record = {};
```
Comment (cindyli) [inline src/framework/core/js/Fluid.js]: Is the `else` block necessary as this situation will be handled at line 1391.

File: src/framework/core/js/Fluid.js [hunk 2]
```diff
@@ -1354,24 +1357,44 @@ var fluid = fluid || fluid_2_0_0;
...
                 }
                 if (!listener) {
                     return;
                 }
+                if (fluid.isPlainObject(listener, true)) {
+                    record = listener;
+                    listener = record.listener;
+                    namespace = record.namespace;
+                    priority = record.priority;
+                    listenerId = record.listenerId;
```
Comment (cindyli) [inline src/framework/core/js/Fluid.js]: Does the `softNamespace` also need to be extracted like what you've done for other listener parts? What I mean is to add a line of ```softNamespace = record.softNamespace```.

File: src/framework/core/js/Fluid.js (hunk 1 above)
Comment (amb26) [inline src/framework/core/js/Fluid.js]: Great catches both, @cindyli - thanks!

**Comment by cindyli** [comment]
Merged at ef4324149aba47693d85f61357feecf6b127a9bf

**Pull request merged**